client/src/main/java/io/pravega/client/stream/impl/ControllerImpl.java

```diff
@@ -254,38 +258,42 @@ public ControllerImpl(ManagedChannelBuilder<?> channelBuilder) {
 
     @Override
     public CompletableFuture<Boolean> scaleStream(final Stream stream, final List<Integer> sealedSegments,
+                                                  final Map<Double, Double> newKeyRanges, final long timeoutMillis,
```

Comment:
I find this timeout parameter odd. In general, I don't think we should ask the application to choose timeout values on public APIs. If we need a timeout, then we should decide it internally. Another option is to enable some form of cancellation and let it run indefinitely and stop upon cancellation.